Linux/arm build warning with unw_getcontext

```
In file included from /__w/1/s/src/coreclr/src/pal/src/exception/seh.cpp:407:
/__w/1/s/src/coreclr/src/pal/src/exception/seh-unwind.cpp:321:10: warning: use of SP or PC in the list is deprecated [-Winline-asm]
    st = unw_getcontext(&unwContext);
         ^
/__w/1/s/artifacts/obj/coreclr/Linux.arm.Debug/src/pal/src/libunwind/include/libunwind-common.h:127:29: note: expanded from macro 'unw_getcontext'
#define unw_getcontext(uc)              unw_tdep_getcontext(uc)
                                        ^
/__w/1/s/src/coreclr/src/pal/src/libunwind/include/libunwind-arm.h:288:6: note: expanded from macro 'unw_tdep_getcontext'
    "stmia %[base], {r0-r15}\n"                                         \
     ^
<inline asm>:5:1: note: instantiated into assembly here
stmia r0, {r0-r15}
^
1 warning generated.
```

This came up before, in https://github.com/dotnet/runtime/issues/5595, where it was stated that the behavior is acceptable/required for libunwind. The fix before was to suppress the warning by avoiding inlining (apparently). Somehow, that fix isn't good enough anymore.

Example: https://dev.azure.com/dnceng/public/_build/results?buildId=712094&view=logs&j=7c52437f-2051-5ead-95de-b55e3296cb61&t=8db66bc6-175f-59c5-6a1e-742f332c2bdf

@janvorli

--- a/src/coreclr/src/pal/src/exception/seh-unwind.cpp
+++ b/src/coreclr/src/pal/src/exception/seh-unwind.cpp
@@ -317,7 +317,18 @@ BOOL PAL_VirtualUnwind(CONTEXT *context, KNONVOLATILE_CONTEXT_POINTERS *contextP
     }
 
 #if !UNWIND_CONTEXT_IS_UCONTEXT_T
+// The unw_getcontext is defined in the libunwind headers for ARM as inline assembly with
+// stmia instruction storing SP and PC, which clang complains about as deprecated.
+// However, it is required for atomic restoration of the context, so disable that warning.
+#if defined(__llvm__) && defined(TARGET_ARM)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winline-asm"
+#endif
     st = unw_getcontext(&unwContext);
+#if defined(__llvm__) && defined(TARGET_ARM)
+#pragma clang diagnostic pop
+#endif
+
     if (st < 0)
     {
         return FALSE;

